server/game/cards/11.3-SoKL/GrowingAmbition.js

```diff
@@ -0,0 +1,62 @@
+const DrawCard = require('../../drawcard');
+
+class GrowingAmbition extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Search deck',
+            phase: 'challenge',
+            cost: ability.costs.payXGold(() => 1, () => this.controller.drawDeck.length),
+            chooseOpponent: true,
+            handler: context => {
+                this.game.promptForDeckSearch(this.controller, {
+                    numToSelect: context.xValue,
+                    onSelect: (player, card) => this.cardSelected(player, card),
+                    source: this
+                });
+
+                this.game.queueSimpleStep(() => {
+                    this.returnCards(context);
+                });
+            }
+        });
+    }
+
+    cardSelected(player, card) {
+        this.chosenCards = this.chosenCards || [];
+        this.chosenCards.push(card);
+        player.removeCardFromPile(card);
+    }
+
+    returnCards(context) {
+        this.game.addMessage('{0} plays {1} to choose {2}, search their deck and place {3} in their discard pile',
+            context.player, this, context.opponent, this.chosenCards);
+
+        for(let card of this.chosenCards) {
+            context.player.moveCard(card, 'discard pile');
+        }
+
+        this.game.promptForSelect(context.opponent, {
+            activePromptTitle: `Select ${context.xValue} card(s)`,
+            cardCondition: card => card.controller === this.controller && card.location === 'discard pile',
+            numCards: context.xValue,
```

Comment:
This should also be an 'exactly' prompt, shouldn't it? The opponent must select the same number of cards discarded, not just up to the number discarded.